cli/upload.py

```diff
@@ -265,38 +279,26 @@ def _gsutil_assay_upload(upload_info: api.UploadInfo, xlsx: str):
         except StopIteration:
             all_uploads_have_run = True
 
-        err = _wait_for_upload(procs, file_count)
-
-        if err:
-            for p in procs:
-
-                if p.poll() != 0:
-
-                    # stopping all other processes
-                    for other_p in procs:
-                        if p != other_p:
-                            other_p.kill()
-
-                    click.echo(
-                        f"\nGCS upload failed on {p.args[-2]} with the following message:\n"
-                    )
-                    click.secho(err, fg="red")
-
-                    raise click.Abort()
+        _wait_for_upload(procs, file_count, upload_info.optional_files, skipped)
 
     click.echo(
         f"[{file_count}/{file_count} done] All files uploaded to GCS and staged for ingestion."
     )
 
+    return upload_info.gcs_file_map
+
 
-def _compose_file_mapping(upload_info: api.UploadInfo, xlsx: str):
+def _compose_file_mapping(
+    upload_info: api.UploadInfo, xlsx: str
+) -> Tuple[Dict[str, str], List[str]]:
     """
     Returns a list of (source_path, target uri) pairs for all 
     the files from the upload info relative to the `work dir` 
     that is xlsx file locaction. If s source_path is a GCS uri,
     it will return it w/o change.
     """
     res = []
+    skipping_files = []
```

Comment:
Renaming `skipping_files` to `missing_optional_files` and `missing_files` to `missing_required_files` might make the distinction between the two clearer.

Reply:
in next commit